great_expectations/experimental/datasources/interfaces.py

```diff
@@ -371,94 +380,88 @@ def execution_engine_type(self) -> Type[ExecutionEngine]:
     # End Abstract Methods
 
 
-class Batch:
-    # Instance variable declarations
-    _datasource: Datasource
-    _data_asset: DataAsset
-    _batch_request: BatchRequest
-    _data: BatchDataType
-    _id: str
+class Batch(ExperimentalBaseModel):
```

Comment:
Why does this need to be an `ExperimentalBaseModel`?

Reply:
We don't serialize it now, although it's possible in the future, since this class does seem to be taking on more importance and we have plans to build a metric store. The biggest pros in my eyes are immutability and default error messages. If you think this means I should define a custom `BaseModel` instead of using `ExperimentalBaseModel` that could be do-able, but I think this makes sense as-is.